### PR TITLE
Add DSA support for ASUS RT-AC58U and Linksys EA6350 v3

### PR DESCRIPTION
**Pull request by giannoug**
The two commits contain changes that convert ASUS RT-AC58U and Linksys EA6350 v3 to DSA. If you can review and merge it will greatly help with testing of these devices.

### CODE REVIEW

File: target/linux/ipq40xx/base-files/etc/board.d/02_network [hunk 1]
```diff
@@ -138,12 +138,6 @@ ipq40xx_setup_macs()
 	8dev,habanero-dvk)
 		label_mac=$(mtd_get_mac_binary "ART" 0x1006)
 		;;
-	asus,rt-ac58u)
-		CI_UBIPART=UBI_DEV
-		wan_mac=$(mtd_get_mac_binary_ubi Factory 0x1006)
-		lan_mac=$(mtd_get_mac_binary_ubi Factory 0x5006)
```
Comment (robimarko) [inline target/linux/ipq40xx/base-files/etc/board.d/02_network]: Well, you cant just drop the MAC setting.
You can use NVMEM for getting the MAC directly in the DTS

File: target/linux/ipq40xx/base-files/etc/board.d/02_network [hunk 2]
```diff
@@ -166,10 +166,6 @@ ipq40xx_setup_macs()
 	ezviz,cs-w3-wd1200g-eup)
 		label_mac=$(mtd_get_mac_binary "ART" 0x6)
 		;;
-	linksys,ea6350v3)
-		wan_mac=$(mtd_get_mac_ascii devinfo hw_mac_addr)
-		lan_mac=$(macaddr_add "$wan_mac" 1)
```
Comment (robimarko) [inline target/linux/ipq40xx/base-files/etc/board.d/02_network]: Same thing here as well

**Comment by giannoug** [comment]
@robimarko I've reverted the changes and rebased. Can you please check now?

**Comment by robimarko** [comment]
Can you check if MAC addresses are set correctly?
Setting them in userspace is a hack, you really should use NVMEM as they are parsed from storage

**Comment by giannoug** [comment]
I'm not totally sure how to do that, but I'm willing to if you can guide me.

The following return the correct MAC addresses, so I can work from there and check:
```
# . /lib/functions/system.sh
# CI_UBIPART=UBI_DEV mtd_get_mac_binary_ubi Factory 0x1006
# CI_UBIPART=UBI_DEV mtd_get_mac_binary_ubi Factory 0x5006
```

I've tried various methods I found inside `02_network` but none of them work 🤔

**Comment by robimarko** [comment]
Well, I actually thought more about this and there isn't a way around it as the MAC is inside of UBI volume and NVMEM wont work.

**Pull request merged**